File: vache.js
const fs = require('fs');
const cowsay = require('cowsay');

function genererVacheHTML(message) {
    const vacheDessin = cowsay.say({ text: message });
    return `
        <!DOCTYPE html>
        <html lang="en">
        <head>
            <meta charset="UTF-8">
            <meta name="viewport" content="width=device-width, initial-scale=1.0">
            <title>Vache Dessinée</title>
        </head>
        <body>
            <pre>${vacheDessin}</pre>
        </body>
        </html>
    `;
}

function ecrireDansFichierHTML(nomFichier, contenu, callback) {
    fs.writeFile(nomFichier, contenu, callback);
}

const messageVache = "Les vaches sont d'excellentes amies, elles vous écoutent toujours !";
const contenuHTML = genererVacheHTML(messageVache);
const nomFichierHTML = "vache_dessinee.html";

ecrireDansFichierHTML(nomFichierHTML, contenuHTML, (err) => {
    if (err) {
        console.error("Erreur lors de l'écriture dans le fichier HTML :", err);
    } else {
        console.log(`Fichier HTML avec une vache dessinée a été créé : ${nomFichierHTML}`);
    }
});

File: vache.py
import cowsay
import random
import sys


def parler_de_vache():
    phrases = [
        "Les vaches sont d'excellentes amies, elles vous écoutent toujours !",
        "Pourquoi les vaches ne peuvent-elles pas être secrètes ? Parce qu'elles laissent toujours des empreintes !",
        "Quel est le plat préféré d'une vache ? L'herbe à la menthe !",
        "Comment appelez-vous une vache qui danse ? Une vache-laitière !",
        "Pourquoi les vaches ne peuvent-elles pas utiliser un ordinateur ? Parce qu'elles ont du mal à saisir la souris !",
    ]
    return random.choice(phrases)


def ecrire_fichier(nom_fichier, texte):
    """ecrire le print cowsay dans un fichier.txt"""
    with open(nom_fichier, "w") as f:
        # We redirect the 'sys.stdout' command towards the descriptor file
        sys.stdout = f
        cowsay.cow(texte)
        # We reset the 'sys.stdout' command
        sys.stdout = sys.__stdout__


if __name__ == "__main__":
    message_vache = parler_de_vache()
    ecrire_fichier("paroles_vache.txt", message_vache)
